include/envoy/buffer/buffer.h

```diff
@@ -189,6 +189,14 @@ class Instance {
    */
   virtual ssize_t search(const void* data, uint64_t size, size_t start) const PURE;
 
+  /**
+   * Search for an occurrence of a buffer at the start of a larger buffer.
+   * @param data supplies the data to search for.
+   * @param size supplies the length of the data to search for.
+   * @return true if this buffer starts with data, false otherwise.
+   */
+  virtual bool startsWith(const void* data, uint64_t size) const PURE;
```

Comment:
prefer `absl::string_view` instead of {data, size}, we do have const void* and uint64 pair from days before absl though we should use string_view for new methods.

Reply:
Even in Buffer?

Reply:
Yes we already do this for `add` and `prepend` (prepend is new so it doesn't have const void*, uint64_t variant)